src/core-api/user/user.service.spec.ts

```
@@ -0,0 +1,158 @@
+                entity: User,
+                // eslint-disable-next-line @typescript-eslint/require-await
+              ): Promise<User> => {
+                if (userRepoSaveException) {
+                  throw new BadRequestException();
+                }
+
+                return user;
+              },
+            ),
+            create: jest.fn(
+              // eslint-disable-next-line @typescript-eslint/no-unused-vars
+              (entityLike: DeepPartial<User>): User => user,
+            ),
+          },
+        },
+        UserService,
+        UserMapper,
+      ],
+    }).compile();
+
+    userService = moduleRef.get(UserService);
+    userRepository = moduleRef.get(UserRepository);
+  });
+
+  beforeEach(() => {
+    userRepoSaveException = false;
+    userRepoFindException = false;
+    user = {
+      id: 1,
+      email: 'test@test.com',
+      phoneNumber: '+38 (096) 123-12-12',
+      firstName: 'test',
+      lastName: 'case',
+    } as User;
+  });
+
+  describe('create', () => {
+    it('should create a new user', async () => {
+      const actualResult = await userService.create({
+        firstName: 'test',
+        lastName: 'case',
+        email: 'test@test.com',
+        phoneNumber: '+380961231212',
+      });
+
+      expect(actualResult).toEqual(user);
+    });
+
+    it('should throw BadRequestException', async () => {
+      userRepoSaveException = true;
+      const actualResult = userService.create({
+        firstName: 'test',
+        lastName: 'case',
+        email: 'test@test.com',
+        phoneNumber: '+380961231212',
+      });
+
+      await expect(actualResult).rejects.toThrowError(BadRequestException);
+    });
+  });
+
+  describe('update', () => {
+    it('should update the user', async () => {
+      const actualResult = await userService.update(1, {
+        lastName: 'jest',
+      });
+
+      expect(actualResult).toEqual(user);
+    });
+
+    it('should throw NotFoundException', async () => {
```

should throw exception if user not found